Index: c/src/lib/libbsp/sparc64/shared/include/traptable.h
===================================================================
RCS file: /usr1/CVS/rtems/c/src/lib/libbsp/sparc64/shared/include/traptable.h,v
retrieving revision 1.2
diff -u -p -r1.2 traptable.h
--- c/src/lib/libbsp/sparc64/shared/include/traptable.h	26 Aug 2010 18:04:51 -0000	1.2
+++ c/src/lib/libbsp/sparc64/shared/include/traptable.h	15 Nov 2010 22:14:33 -0000
@@ -14,9 +14,10 @@
 
 #define ENTRY_SIZE (32)
 
-#define TABLE_SIZE (1024*ENTRY_SIZE * 2 /* HACK */ )
+#define TABLE_SIZE (1024*ENTRY_SIZE)
 
 #ifndef ASM
+extern void* real_trap_table;
 extern void* trap_table[TABLE_SIZE];
 #endif
 
Index: c/src/lib/libbsp/sparc64/shared/start/trap_table.S
===================================================================
RCS file: /usr1/CVS/rtems/c/src/lib/libbsp/sparc64/shared/start/trap_table.S,v
retrieving revision 1.1
diff -u -p -r1.1 trap_table.S
--- c/src/lib/libbsp/sparc64/shared/start/trap_table.S	17 Jun 2010 16:20:45 -0000	1.1
+++ c/src/lib/libbsp/sparc64/shared/start/trap_table.S	15 Nov 2010 22:14:33 -0000
@@ -11,194 +11,147 @@
  * 
  *  This code defines the space for the trap table used by sun4u.
  *
- *  This file is currently unused.
- *
  * $Id: trap_table.S,v 1.1 2010/06/17 16:20:45 joel Exp $
  */
 
-
 #include <rtems/asm.h>
 #include <rtems/score/cpu.h>
 #include <traptable.h>
 
-#define BAD_TRAP \
-    ta  0; \
-  nop; \
-  nop; \
-  nop; \
-  nop; \
-  nop; \
-  nop; \
+.register %g2, #scratch
+
+.macro JUMP_TO_TRAP offset
+  sethi %hi(real_trap_table), %g1;
+  sethi %hi(\offset), %g2;
+  or    %g1, %lo(real_trap_table), %g1;
+  or    %g2, %lo(\offset), %g2;          ! g2 = offset
+  ldx   [%g1], %g1;                      ! g1 = real_trap_table
+  add   %g1, %g2, %g1;                   ! g1 = real_trap_table[offset]
+  jmpl  %g1, %g0;
   nop;
+.endm
 
-.section .ttable
+! space to save a pointer to the real trap table
+.section  .data
+  .align 8
+  .global real_trap_table
+  SYM(real_trap_table):
+  .space 8
+
+.section .text
   .align (TABLE_SIZE)
 PUBLIC(trap_table)
   SYM(trap_table):
-  .space TABLE_SIZE
-
-/* These are just bogus traps that aren't used ever
 
-.org trap_table + 8*32
-    BAD_TRAP;                                     ! 8 IAE
-.org trap_table + 9*32
-    BAD_TRAP;                                     ! 9
-.org trap_table + 10*32
-    BAD_TRAP;                                     ! A
-
-.org trap_table + 11*32
-    BAD_TRAP;                                     ! B
-.org trap_table + 12*32
-    BAD_TRAP;                                     ! C
-
-.org trap_table + 16*32
-    BAD_TRAP;                                     ! 10 illegal instruction
-.org trap_table + 17*32
-    BAD_TRAP;                                     ! 11 privileged op_code
-
-.org trap_table + 18*32
-    BAD_TRAP;                                     ! 12 unimplemented LDD
-.org trap_table + 19*32
-    BAD_TRAP;                                     ! 13 unimplemented std
-.org trap_table + 20*32
-    BAD_TRAP;                                     ! 14 dae invalid asi
-.org trap_table + 21*32
-    BAD_TRAP;                                     ! 15 dae priv violation
-.org trap_table + 22*32
-    BAD_TRAP;                                     ! 16 dae nc page
-.org trap_table + 23*32
-    BAD_TRAP;                                     ! 17 dae nfo page
-
-.org trap_table + 32*32
-    BAD_TRAP;                                     ! 20 fp_disabled
-.org trap_table + 33*32
-    BAD_TRAP;                                     ! 21 fp_exception_ieee_754
-.org trap_table + 34*32
-    BAD_TRAP;                                     ! 22 fp_exception_other
-
-.org trap_table + 35*32
-    BAD_TRAP;                                     ! 23 tag_overflow
-
-.org trap_table + 36*32
-    BAD_TRAP;                         ! 24-27 clean_window
-
-
-.org trap_table + 40*32
-    BAD_TRAP;                                     ! 28 div by zero
-    
-.org trap_table + 48*32
-    BAD_TRAP;                                     ! 30 data access exception
-
-.org trap_table + 49*32
-    BAD_TRAP;                                     ! 31 data mmu miss
-.org trap_table + 50*32
-    BAD_TRAP;                                     ! 32 data access error
-
-.org trap_table + 52*32
-    BAD_TRAP;                                     ! 34 mem address not aligned 
-.org trap_table + 53*32
-    BAD_TRAP;                                     ! 35 LDDF mem address not aligned 
-.org trap_table + 54*32
-    BAD_TRAP;                                     ! 36 STDF mem addr not aligned
-.org trap_table + 55*32
-    BAD_TRAP;                                     ! 37 privileged action
-.org trap_table + 56*32
-    BAD_TRAP;                                     ! 38 LDQF mem addr not aligned
-.org trap_table + 57*32
-    BAD_TRAP;                                     ! 39 STQF mem addr not aligned
-
-
-.org trap_table + 65*32
-    BAD_TRAP;                                     ! 41-4F
-                                                  ! interrupt_level_n (1-15) 
-.org trap_table + 66*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 67*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 68*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 69*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 70*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 71*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 72*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 73*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 74*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 75*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 76*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 77*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 78*32
-    BAD_TRAP;                                     ! 
-.org trap_table + 79*32
-    BAD_TRAP;                                     ! 
-
-.org trap_table + 124*32
-    BAD_TRAP;                                     ! 7C cpu mondo
-
-.org trap_table + 125*32
-    BAD_TRAP;                                     ! 7D dev mondo
-
-.org trap_table + 128*32
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 80-83 spill_0_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 84-87 spill_1_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 88-8B spill_2_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 8C-8F spill_3_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 90-93 spill_4_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 94-97 spill_5_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 98-9B spill_6_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! 9C-9F spill_7_normal
-
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! A0-A3 spill_0_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! A4-A7 spill_1_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! A8-AB spill_2_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! AC-AF spill_3_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! B0-B3 spill_4_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! B4-B7 spill_5_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! B8-BB spill_6_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! BC-BF spill_7_other
-
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! C0-C3 fill_0_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! C4-C7 fill_1_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! C8-CB fill_2_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! CC-CF fill_3_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! D0-D3 fill_4_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! D4-D7 fill_5_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! D8-DB fill_6_normal
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! DC-DF fill_7_normal
-
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! E0-E3 fill_0_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! E4-E7 fill_1_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! E8-EB fill_2_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! EC-EF fill_3_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! F0-F3 fill_4_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! F4-F7 fill_5_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! F8-FB fill_6_other
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;          ! FC-FF fill_7_other
-
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;                   ! 100-17F trap instruction
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;                                    
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-    BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;BAD_TRAP;
-*/
+  .irp idx, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, \
+    11, 12, 13, 14, 15, 16, 17, 18, 19, 20, \
+    21, 22, 23, 24, 25, 26, 27, 28, 29, 30, \
+    31, 32, 33, 34, 35, 36, 37, 38, 39, 40, \
+    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, \
+    51, 52, 53, 54, 55, 56, 57, 58, 59, 60, \
+    61, 62, 63, 64, 65, 66, 67, 68, 69, 70, \
+    71, 72, 73, 74, 75, 76, 77, 78, 79, 80, \
+    81, 82, 83, 84, 85, 86, 87, 88, 89, 90, \
+    91, 92, 93, 94, 95, 96, 97, 98, 99, 100, \
+    101, 102, 103, 104, 105, 106, 107, 108, 109, 110, \
+    111, 112, 113, 114, 115, 116, 117, 118, 119, 120, \
+    121, 122, 123, 124, 125, 126, 127, 128, 129, 130, \
+    131, 132, 133, 134, 135, 136, 137, 138, 139, 140, \
+    141, 142, 143, 144, 145, 146, 147, 148, 149, 150, \
+    151, 152, 153, 154, 155, 156, 157, 158, 159, 160, \
+    161, 162, 163, 164, 165, 166, 167, 168, 169, 170, \
+    171, 172, 173, 174, 175, 176, 177, 178, 179, 180, \
+    181, 182, 183, 184, 185, 186, 187, 188, 189, 190, \
+    191, 192, 193, 194, 195, 196, 197, 198, 199, 200, \
+    201, 202, 203, 204, 205, 206, 207, 208, 209, 210, \
+    211, 212, 213, 214, 215, 216, 217, 218, 219, 220, \
+    221, 222, 223, 224, 225, 226, 227, 228, 229, 230, \
+    231, 232, 233, 234, 235, 236, 237, 238, 239, 240, \
+    241, 242, 243, 244, 245, 246, 247, 248, 249, 250, \
+    251, 252, 253, 254, 255, 256, 257, 258, 259, 260, \
+    261, 262, 263, 264, 265, 266, 267, 268, 269, 270, \
+    271, 272, 273, 274, 275, 276, 277, 278, 279, 280, \
+    281, 282, 283, 284, 285, 286, 287, 288, 289, 290, \
+    291, 292, 293, 294, 295, 296, 297, 298, 299, 300, \
+    301, 302, 303, 304, 305, 306, 307, 308, 309, 310, \
+    311, 312, 313, 314, 315, 316, 317, 318, 319, 320, \
+    321, 322, 323, 324, 325, 326, 327, 328, 329, 330, \
+    331, 332, 333, 334, 335, 336, 337, 338, 339, 340, \
+    341, 342, 343, 344, 345, 346, 347, 348, 349, 350, \
+    351, 352, 353, 354, 355, 356, 357, 358, 359, 360, \
+    361, 362, 363, 364, 365, 366, 367, 368, 369, 370, \
+    371, 372, 373, 374, 375, 376, 377, 378, 379, 380, \
+    381, 382, 383, 384, 385, 386, 387, 388, 389, 390, \
+    391, 392, 393, 394, 395, 396, 397, 398, 399, 400, \
+    401, 402, 403, 404, 405, 406, 407, 408, 409, 410, \
+    411, 412, 413, 414, 415, 416, 417, 418, 419, 420, \
+    421, 422, 423, 424, 425, 426, 427, 428, 429, 430, \
+    431, 432, 433, 434, 435, 436, 437, 438, 439, 440, \
+    441, 442, 443, 444, 445, 446, 447, 448, 449, 450, \
+    451, 452, 453, 454, 455, 456, 457, 458, 459, 460, \
+    461, 462, 463, 464, 465, 466, 467, 468, 469, 470, \
+    471, 472, 473, 474, 475, 476, 477, 478, 479, 480, \
+    481, 482, 483, 484, 485, 486, 487, 488, 489, 490, \
+    491, 492, 493, 494, 495, 496, 497, 498, 499, 500, \
+    501, 502, 503, 504, 505, 506, 507, 508, 509, 510, 511
+
+    .org trap_table + ((\idx)*32)
+      JUMP_TO_TRAP( ((\idx)*32) );
+  .endr
+
+  .irp idx, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, \
+    11, 12, 13, 14, 15, 16, 17, 18, 19, 20, \
+    21, 22, 23, 24, 25, 26, 27, 28, 29, 30, \
+    31, 32, 33, 34, 35, 36, 37, 38, 39, 40, \
+    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, \
+    51, 52, 53, 54, 55, 56, 57, 58, 59, 60, \
+    61, 62, 63, 64, 65, 66, 67, 68, 69, 70, \
+    71, 72, 73, 74, 75, 76, 77, 78, 79, 80, \
+    81, 82, 83, 84, 85, 86, 87, 88, 89, 90, \
+    91, 92, 93, 94, 95, 96, 97, 98, 99, 100, \
+    101, 102, 103, 104, 105, 106, 107, 108, 109, 110, \
+    111, 112, 113, 114, 115, 116, 117, 118, 119, 120, \
+    121, 122, 123, 124, 125, 126, 127, 128, 129, 130, \
+    131, 132, 133, 134, 135, 136, 137, 138, 139, 140, \
+    141, 142, 143, 144, 145, 146, 147, 148, 149, 150, \
+    151, 152, 153, 154, 155, 156, 157, 158, 159, 160, \
+    161, 162, 163, 164, 165, 166, 167, 168, 169, 170, \
+    171, 172, 173, 174, 175, 176, 177, 178, 179, 180, \
+    181, 182, 183, 184, 185, 186, 187, 188, 189, 190, \
+    191, 192, 193, 194, 195, 196, 197, 198, 199, 200, \
+    201, 202, 203, 204, 205, 206, 207, 208, 209, 210, \
+    211, 212, 213, 214, 215, 216, 217, 218, 219, 220, \
+    221, 222, 223, 224, 225, 226, 227, 228, 229, 230, \
+    231, 232, 233, 234, 235, 236, 237, 238, 239, 240, \
+    241, 242, 243, 244, 245, 246, 247, 248, 249, 250, \
+    251, 252, 253, 254, 255, 256, 257, 258, 259, 260, \
+    261, 262, 263, 264, 265, 266, 267, 268, 269, 270, \
+    271, 272, 273, 274, 275, 276, 277, 278, 279, 280, \
+    281, 282, 283, 284, 285, 286, 287, 288, 289, 290, \
+    291, 292, 293, 294, 295, 296, 297, 298, 299, 300, \
+    301, 302, 303, 304, 305, 306, 307, 308, 309, 310, \
+    311, 312, 313, 314, 315, 316, 317, 318, 319, 320, \
+    321, 322, 323, 324, 325, 326, 327, 328, 329, 330, \
+    331, 332, 333, 334, 335, 336, 337, 338, 339, 340, \
+    341, 342, 343, 344, 345, 346, 347, 348, 349, 350, \
+    351, 352, 353, 354, 355, 356, 357, 358, 359, 360, \
+    361, 362, 363, 364, 365, 366, 367, 368, 369, 370, \
+    371, 372, 373, 374, 375, 376, 377, 378, 379, 380, \
+    381, 382, 383, 384, 385, 386, 387, 388, 389, 390, \
+    391, 392, 393, 394, 395, 396, 397, 398, 399, 400, \
+    401, 402, 403, 404, 405, 406, 407, 408, 409, 410, \
+    411, 412, 413, 414, 415, 416, 417, 418, 419, 420, \
+    421, 422, 423, 424, 425, 426, 427, 428, 429, 430, \
+    431, 432, 433, 434, 435, 436, 437, 438, 439, 440, \
+    441, 442, 443, 444, 445, 446, 447, 448, 449, 450, \
+    451, 452, 453, 454, 455, 456, 457, 458, 459, 460, \
+    461, 462, 463, 464, 465, 466, 467, 468, 469, 470, \
+    471, 472, 473, 474, 475, 476, 477, 478, 479, 480, \
+    481, 482, 483, 484, 485, 486, 487, 488, 489, 490, \
+    491, 492, 493, 494, 495, 496, 497, 498, 499, 500, \
+    501, 502, 503, 504, 505, 506, 507, 508, 509, 510, 511
+
+    .org trap_table + 512*32 + ((\idx)*32)
+      JUMP_TO_TRAP( 512*32 + ((\idx)*32) );
+  .endr
 
Index: c/src/lib/libbsp/sparc64/shared/startup/linkcmds
===================================================================
RCS file: /usr1/CVS/rtems/c/src/lib/libbsp/sparc64/shared/startup/linkcmds,v
retrieving revision 1.2
diff -u -p -r1.2 linkcmds
--- c/src/lib/libbsp/sparc64/shared/startup/linkcmds	3 Aug 2010 18:29:12 -0000	1.2
+++ c/src/lib/libbsp/sparc64/shared/startup/linkcmds	15 Nov 2010 22:14:33 -0000
@@ -202,11 +202,6 @@ SECTIONS
      PROVIDE (__stack = .);
      PROVIDE (StackStart = .);
   } > ram
-
-  /* made-up section for the trap-table, which has a very high alignment */
-  . = ALIGN(1024*32);
-  .ttable   : { *(.ttable) } > ram
-  
 }
 
 
Index: c/src/lib/libbsp/sparc64/usiii/start/bspinit.S
===================================================================
RCS file: /usr1/CVS/rtems/c/src/lib/libbsp/sparc64/usiii/start/bspinit.S,v
retrieving revision 1.1
diff -u -p -r1.1 bspinit.S
--- c/src/lib/libbsp/sparc64/usiii/start/bspinit.S	17 Jun 2010 16:20:39 -0000	1.1
+++ c/src/lib/libbsp/sparc64/usiii/start/bspinit.S	15 Nov 2010 22:14:33 -0000
@@ -35,12 +35,17 @@ PUBLIC(_BSP_init)
   SYM(_BSP_init):
   save %sp, -STACK_WINDOW_SAVE_AREA_SIZE, %sp
 
-  /* copy firmware trap table so that RTEMS can install ISR handlers */
-  setx  SYM(trap_table), %l0, %o0
-  rdpr  %tba, %o1
-  set TABLE_SIZE, %o2
-  call memcpy
-  nop
+  ! copy the tba
+  sethi %hi(real_trap_table), %o0
+  rdpr  %tba, %o2
+  stx   %o2, [%o0 + %lo(real_trap_table)]
+
+!  /* copy firmware trap table so that RTEMS can install ISR handlers */
+!  setx  SYM(trap_table), %l0, %o0
+!  rdpr  %tba, %o1
+!  set TABLE_SIZE, %o2
+!  call memcpy
+!  nop
 
   mov %g0, %o0
   call _take_mmu
