diff -Naur binutils-2.20-original/bfd/config.bfd binutils-2.20-rtems/bfd/config.bfd
--- binutils-2.20-original/bfd/config.bfd	2009-08-06 13:38:00.000000000 -0400
+++ binutils-2.20-rtems/bfd/config.bfd	2010-04-26 15:33:50.000000000 -0400
@@ -1430,7 +1430,7 @@
     targ_selvecs="bfd_elf32_sparc_vec sparclinux_vec sunos_big_vec"
     want64=true
     ;;
-  sparc64-*-elf*)
+  sparc64-*-elf* | sparc64-*-rtems* )
     targ_defvec=bfd_elf64_sparc_vec
     targ_selvecs=bfd_elf32_sparc_vec
     want64=true
@@ -1439,7 +1439,7 @@
   sparc*-*-coff*)
     targ_defvec=sparccoff_vec
     ;;
-  sparc*-*-rtems*)
+  sparc-*-rtems*)
     targ_defvec=bfd_elf32_sparc_vec
     targ_selvecs="sunos_big_vec sparccoff_vec"
     ;;
diff -Naur binutils-2.20-original/ld/configure.tgt binutils-2.20-rtems/ld/configure.tgt
--- binutils-2.20-original/ld/configure.tgt	2009-08-06 13:38:03.000000000 -0400
+++ binutils-2.20-rtems/ld/configure.tgt	2010-04-26 15:35:41.000000000 -0400
@@ -564,6 +564,7 @@
 			targ_extra_libpath=$targ_extra_emuls ;;
 sparc64-*-aout*)	targ_emul=sparcaout ;;
 sparc64-*-elf*)		targ_emul=elf64_sparc ;;
+sparc64-*-rtems*)	targ_emul=elf64_sparc ;;
 sparc-sun-sunos4*) 	targ_emul=sun4 ;;
 sparclite*-*-elf)	targ_emul=elf32_sparc ;;
 sparclite*-*-coff)	targ_emul=coff_sparc ;;
@@ -612,7 +613,7 @@
 			tdir_elf32_sparc=`echo ${targ_alias} | sed -e 's/64//'` ;;
 sparc*-*-solaris2*)	targ_emul=elf32_sparc ;;
 sparc*-wrs-vxworks*)	targ_emul=sparcaout ;;
-sparc*-*-rtems*)        targ_emul=elf32_sparc
+sparc-*-rtems*)		targ_emul=elf32_sparc
 			;;
 spu-*-elf*)		targ_emul=elf32_spu ;;
 tic30-*-*aout*)		targ_emul=tic30aout ;;
