diff -Naur binutils-2.19/bfd/config.bfd binutils-2.19-original/bfd/config.bfd
--- binutils-2.19/bfd/config.bfd	2009-10-15 12:06:37.000000000 -0400
+++ binutils-2.19-original/bfd/config.bfd	2008-04-16 14:02:02.000000000 -0400
@@ -1397,7 +1397,7 @@
     targ_selvecs="bfd_elf32_sparc_vec sparclinux_vec sunos_big_vec"
     want64=true
     ;;
-  sparc64-*-elf* | sparc64-*-rtems* )
+  sparc64-*-elf*)
     targ_defvec=bfd_elf64_sparc_vec
     targ_selvecs=bfd_elf32_sparc_vec
     want64=true
@@ -1406,7 +1406,7 @@
   sparc*-*-coff*)
     targ_defvec=sparccoff_vec
     ;;
-  sparc-*-rtems*)
+  sparc*-*-rtems*)
     targ_defvec=bfd_elf32_sparc_vec
     targ_selvecs="sunos_big_vec sparccoff_vec"
     ;;
diff -Naur binutils-2.19/ld/configure.tgt binutils-2.19-original/ld/configure.tgt
--- binutils-2.19/ld/configure.tgt	2009-10-15 12:07:52.000000000 -0400
+++ binutils-2.19-original/ld/configure.tgt	2008-08-09 01:35:14.000000000 -0400
@@ -550,7 +550,6 @@
 			targ_extra_libpath=$targ_extra_emuls ;;
 sparc64-*-aout*)	targ_emul=sparcaout ;;
 sparc64-*-elf*)		targ_emul=elf64_sparc ;;
-sparc64-*-rtems*)	targ_emul=elf64_sparc ;;
 sparc-sun-sunos4*) 	targ_emul=sun4 ;;
 sparclite*-*-elf)	targ_emul=elf32_sparc ;;
 sparclite*-*-coff)	targ_emul=coff_sparc ;;
@@ -599,7 +598,7 @@
 			tdir_elf32_sparc=`echo ${targ_alias} | sed -e 's/64//'` ;;
 sparc*-*-solaris2*)	targ_emul=elf32_sparc ;;
 sparc*-wrs-vxworks*)	targ_emul=sparcaout ;;
-sparc-*-rtems*)        targ_emul=elf32_sparc
+sparc*-*-rtems*)        targ_emul=elf32_sparc
 			;;
 spu-*-elf*)		targ_emul=elf32_spu ;;
 tic30-*-*aout*)		targ_emul=tic30aout ;;
