Index: cpukit/score/src/objectextendinformation.c
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/score/src/objectextendinformation.c,v
retrieving revision 1.22
diff -u -p -r1.22 objectextendinformation.c
--- cpukit/score/src/objectextendinformation.c	21 Jun 2010 16:27:37 -0000	1.22
+++ cpukit/score/src/objectextendinformation.c	31 Aug 2010 17:08:06 -0000
@@ -142,9 +142,10 @@ void _Objects_Extend_information(
     /*
      *  Allocate the tables and break it up.
      */
-    block_size = block_count *
-           (sizeof(void *) + sizeof(uint32_t) + sizeof(Objects_Name *)) +
-          ((maximum + minimum_index) * sizeof(Objects_Control *));
+    block_size = (block_count * (sizeof(void *) + sizeof(uint32_t)) + 
+           CPU_ALIGNMENT - 1) & ~(CPU_ALIGNMENT - 1);
+    block_size += block_count * sizeof(Objects_Name *) +
+           ((maximum + minimum_index) * sizeof(Objects_Control *));
     object_blocks = (void**) _Workspace_Allocate( block_size );
 
     if ( !object_blocks ) {
@@ -160,6 +161,9 @@ void _Objects_Extend_information(
     local_table = (Objects_Control **) _Addresses_Add_offset(
         inactive_per_block, block_count * sizeof(uint32_t) );
 
+    local_table = (Objects_Control **)
+        ((((uintptr_t)local_table) + CPU_ALIGNMENT - 1) & ~(CPU_ALIGNMENT - 1));
+
     /*
      *  Take the block count down. Saves all the (block_count - 1)
      *  in the copies.
