diff -Naur gcc-4.4.2/gcc/config/sparc/sp64-rtemself.h gcc-4.4.2-rtems/gcc/config/sparc/sp64-rtemself.h
--- gcc-4.4.2/gcc/config/sparc/sp64-rtemself.h	1969-12-31 19:00:00.000000000 -0500
+++ gcc-4.4.2-rtems/gcc/config/sparc/sp64-rtemself.h	2010-04-26 15:27:18.000000000 -0400
@@ -0,0 +1,37 @@
+/* Definitions for rtems targeting a SPARC64 using ELF.
+   Copyright (C) 2009 Free Software Foundation, Inc.
+   Contributed by Joel Sherrill (joel@OARcorp.com).
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+/* Target OS builtins.  */
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()        \
+  do                        \
+    {                        \
+    builtin_define ("__rtems__");        \
+    builtin_define ("__USE_INIT_FINI__");    \
+    builtin_assert ("system=rtems");    \
+    }                        \
+  while (0)
+
+/* Use the default */
+#undef LINK_GCC_C_SEQUENCE_SPEC
+
+/* we are not svr4 */
+#undef CPP_SUBTARGET_SPEC
+#define CPP_SUBTARGET_SPEC ""
diff -Naur gcc-4.4.2/gcc/config.gcc gcc-4.4.2-rtems/gcc/config.gcc
--- gcc-4.4.2/gcc/config.gcc	2010-04-26 15:25:33.000000000 -0400
+++ gcc-4.4.2-rtems/gcc/config.gcc	2010-04-26 15:11:46.000000000 -0400
@@ -2309,6 +2309,12 @@
 	tmake_file="${tmake_file} sparc/t-crtfm"
 	extra_parts="crtbegin.o crtend.o"
 	;;
+sparc64-*-rtems*)
+	tm_file="${tm_file} dbxelf.h elfos.h svr4.h sparc/sysv4.h sparc/sp64-elf.h sparc/sp64-rtemself.h rtems.h"
+	extra_options="${extra_options} sparc/little-endian.opt"
+	tmake_file="${tmake_file} sparc/t-crtfm t-rtems"
+	extra_parts="crtbegin.o crtend.o"
+	;;	
 sparc64-*-freebsd*|ultrasparc-*-freebsd*)
 	tm_file="${tm_file} ${fbsd_tm_file} dbxelf.h elfos.h sparc/sysv4.h sparc/freebsd.h"
 	extra_options="${extra_options} sparc/long-double-switch.opt"
diff -Naur gcc-4.4.2/libgcc/config.host gcc-4.4.2-rtems/libgcc/config.host
--- gcc-4.4.2/libgcc/config.host	2010-04-26 15:25:33.000000000 -0400
+++ gcc-4.4.2-rtems/libgcc/config.host	2010-04-26 15:19:52.000000000 -0400
@@ -542,7 +542,7 @@
 	extra_parts="$extra_parts crtfastmath.o"
 	tmake_file="${tmake_file} sparc/t-crtfm"
 	;;
-sparc-*-rtems*)
+sparc-*-rtems* | sparc64-*-rtems*)
 	;;
 sparc64-*-solaris2* | sparcv9-*-solaris2*)
 	;;
